scripts/delete_resources.sh

```diff
@@ -1,149 +1,240 @@
+                ]
+            }
+        }
+    ]
+}" > payload.json
+      cat payload.json
+      echo "Deleting Route53 record ${NAME} for prefix $1"
+      CHANGE_STATUS=$(aws route53 change-resource-record-sets --hosted-zone-id "${R53_ZONE_ID}" --change-batch file://"${PWD}"/payload.json)
+      echo "CHANGE STATUS: 
+      $CHANGE_STATUS"
+      STATUS_ID=$(echo "$CHANGE_STATUS" | grep Id: | cut -d " " -f 4)
+      echo "Status ID: $STATUS_ID"
+      #Get status from the change
+      aws route53 wait resource-record-sets-changed --id "$STATUS_ID"
+      echo "Successfully deleted Route53 record ${NAME}: status: ${STATUS_ID}"
+      exit
+    done
+    rm -rf "${PWD}"/payload.json
   fi
+}
+
+delete_all_resources () {
+  delete_ec2_instances "$1"
+  delete_db_resources "$1"
+  delete_lb_resources "$1"
+  delete_target_groups "$1"
+  delete_route_s3 "$1"
+}
+
+
+while getopts r:h OPTION
+do 
+  case "${OPTION}"
+    in
+    r) RESOURCES=${OPTARG};;
+    h|?)
+      echo "
+    Usage: 
+      $(basename "$0") [-r <resource_prefix>] [-h]
+
+      -r: resource prefix names. can be comma separated. 
+      -h: help. Prints usage example.
+      
+      Ex 1: 
+      ./delete_jenkins_resources.sh -r resourceprefix1,resourceprefix2,resourceprefix2
+      - This will delete all resources for the names resourceprefix1,resourceprefix2 and resourceprefix2
+      Ex 2:
+      ./delete_jenkins_resources.sh
+      - This will lookup the resource name from local tfvars file and delete the resources. This is interactive and will ask for confirmation before delete
+      Ex 3: 
+      ./delete_jenkins_resources.sh -h
+      - Print usage details for reference.
+      "
+      exit 1
+      ;;
+  esac
+done
+
+if [ "${RESOURCES}" = "" ]; then
+    echo "Working with local .env and .tfvars file to get resource name for deletion"
+    #Get resource name from tfvarslocal && change name to make more sense in this context
+    PRODUCT_NAME=$(grep ENV_PRODUCT <./config/.env | cut -d= -f2 | tr -d ' "')  # Split string based on delimiter =
+    if echo "${PRODUCT_NAME}" | grep -q "ENV_PRODUCT"; then
+      PRODUCT_NAME=$(echo "${PRODUCT_NAME}" | cut -d ":" -f 2)  # Split string based on delimiter :
+    fi
+    if [[ -z "$PRODUCT_NAME" || ! "$PRODUCT_NAME" =~ ^(rke2|k3s)$ ]]; then
+      echo "Wrong or empty product name found in .env file for: $PRODUCT_NAME"
+      exit 1
+    fi
+
+    #Validate path to the tfvars file
+    if [[ ! -f ./config/"$PRODUCT_NAME".tfvars ]]; then
+      echo "No $PRODUCT_NAME.tfvars file found in config directory"
+      exit 1
+    fi
+
+    #Get resource name from tfvars file and validate
+    RESOURCE_NAME=$(grep resource_name <./config/"$PRODUCT_NAME".tfvars | cut -d= -f2 | tr -d ' "')
+    if [[ -z "$RESOURCE_NAME" ]]; then
+      echo "No resource name found for: $PRODUCT_NAME.tfvars file"
+      exit 1
+    fi
+
+    printf "This is going to delete all AWS resources with the prefix %s. Continue (yes/no)? " "$RESOURCE_NAME"
```

Comment:
we also need to fix that when i call it with resource name , its not prompting me if i want to proceed with this name, its just going wild and free and go do his stuff. this is too dangerous.
```
 bash scripts/delete_resources.sh -r ANYNAME

## For prefix name: ANYNAME:
No ec2 instances found with prefix: ANYNAME. Nothing to delete.
No db instances found with prefix ANYNAME. Nothing to delete.
No db clusters found with prefix ANYNAME. Nothing to delete.


```

Reply:
if by any change i do , locally or jenkins 
```
 bash scripts/delete_resources.sh -r a
```

it will delete a bunch of stuff from other people also right

Reply:
uhmm... how do u make jenkins job interactive? like you would go onto the console page and then enter yes on that page?

Reply:
we could potentially have a dry run option maybe? get a param from jenkins - say dry run true, and in here, we could add another var -d, which will only maybe print the resource list and not do the actual delete?

Reply:
from all my reading, jenkins can take in another param from you - say "yes" and pass that via jenkinsfile to the interactive script say ./delete_resources.sh -r abc << input". this is of no use to us, if we are going to provide a yes anyway when we want to delete.

Reply:
honestly idk, but at least in jenkins we have a momentary history of the run and etc...locally we dont.
at least the script needs to check if , running != jenkins, prompt the question first.

in jenkins then its mandatory to run with `RESOURCE_PREFIX_LIST` ?

Reply:
yes - it is mandatory to run with RESOURCE_PREFIX_LIST having a value in jenkins. if you provided empty value, the job will abort. when you provide -r value locally or in the jenkins - you are manually typing the value. we are not reading something from a place you might not know about. this is an intentional choice on a person to go type the exact value/letters they want to delete.

Reply:
if you run the script without providing the -r option - that still does the yes/no check for confirmation. that is the only place when you are not aware of what the prefix name is. since you did not manually type this and ask for deletion.

Reply:
added the dry-run option as well.
fixed.